File: 卡码/代码随想录/C++/贪心/02-中等/LeetCode_0406.cpp
//406. 根据身高重建队列
#include <iostream>
#include <vector>
#include <list>
#include <algorithm>
using namespace std;

class Solution {
public:
  vector<vector<int>> reconstructQueue(vector<vector<int>>& people) {
  	sort(people.begin(),people.end(),[](vector<int>& p1,vector<int>& p2){
			if(p1[0]==p2[0]) return p1[1]<p2[1];
			else return p1[0]>p2[0];
		});
		list<vector<int>> q;
		for(vector<int>& p:people){
			int pos=p[1];
			auto it=q.begin();
			while(pos--) it++;
			q.insert(it,p);
		}
		return vector<vector<int>>(q.begin(),q.end());
  }
};

int main(){
  system("pause");
  return 0;
}

File: 卡码/代码随想录/Python/贪心/02-中等/LeetCode_0406.py
#406. 根据身高重建队列
from typing import *

class Solution:
  def reconstructQueue(self, people: List[List[int]]) -> List[List[int]]:
    q:List[List[int]]=[]
    people.sort(key=lambda p:(-p[0],p[1]))
    for p in people:
      q.insert(p[1],p)
    return q